google/cloud/bigquery/_helpers.py

```diff
@@ -67,6 +71,9 @@ def _verify_bq_storage_version():
         )
         raise LegacyBigQueryStorageError(msg)
 
+    if installed_version >= _BQ_STORAGE_OPTIONAL_READ_SESSION_VERSION:
+        _IS_BQ_STORAGE_READ_SESSION_OPTIONAL = True
```

Comment:
This could potentially be tricky, because the value of the `_IS_BQ_STORAGE_READ_SESSION_OPTIONAL` "constant" depends on whether `_verify_bq_storage_version()` has been called or not.

It would IMO be more straightforward if we performed version detection at import time and already make the final decision on the value of `_IS_BQ_STORAGE_READ_SESSION_OPTIONAL` then.

Reply:
We don't import `google.cloud.bigquery_storage` in `_download_table_bqstorage_stream`. Since that function gets a client and other objects passed in to it.

How about we make sure we call `_verify_bq_storage_version()` from `_download_table_bqstorage_stream()` and optimize to not call the version parsing too often?

Reply:
The overhead of parsing a version is most likely negligible compared to the request/response times, I was more concerned that changing a module "constant" might represent a landmine in the "right" constellation of the planets.

Perhaps just adding a comment that this private "constant" is not a constant and can change when calling `_verify_bq_storage_version()` is enough. You know, just to warn the readers about it.

Reply:
Good point. I think there's a way to do this without "constants" that aren't actually constant. I think maybe I'll create a little class to hold our version comparisons.